src/jobflow_remote/cli/job.py

```diff
@@ -182,30 +189,28 @@ def job_info(
     ] = False,
     verbosity: verbosity_opt = 0,
 ) -> None:
-    """Detail information on a specific job."""
-    db_id, job_id = get_job_db_ids(job_db_id, job_index)
-
-    if with_error:
-        out_console.print(
-            "The --with-error option is deprecated and not needed anymore to show error messages",
-            style="yellow",
-        )
+    """Detailed information on a specific job."""
+    if pid is not None and (job_db_id is not None or job_index is not None):
+        raise typer.BadParameter("Cannot specify both job ID/index and process ID")
 
     with loading_spinner():
         jc = get_job_controller()
 
-        if verbosity > 0:
-            job_data = jc.get_job_doc(
-                job_id=job_id,
-                job_index=job_index,
-                db_id=db_id,
-            )
+        if job_db_id is not None or job_index is not None:
+            if verbosity > 0:
+                job_data = jc.get_job_doc(
+                    job_id=job_db_id,
+                    job_index=job_index,
+                )
+            else:
+                job_data = jc.get_job_info(
+                    job_id=job_db_id,
+                    job_index=job_index,
+                )
+        elif pid is not None:
+            job_data = jc.get_job_info_by_pid(pid)
```

Comment:
> I don't think it is worth adding a get_job_doc_by_pid method to JobController as I suppose this would not be a very common occurrence.

i'll slightly push back on this. i think it makes sense to have that function even if just for consistency of impementation when comparing with `job_db_id`. it's nice to be able to directly translate a CLI command to the Python API which would not be possible if we remove `get_job_info_by_pid`. let me know if you prefer it be removed anyway

Reply:
I agree that it is nice to have this kind of mapping. I am just afraid that in the future this could lead to a lot of similar methods based on different query options. However, since this is not the case at the moment I think it can stay like this and be reevaluated if it becomes an issue in the future.